storage/sftp/sftp.go

```
@@ -0,0 +1,206 @@
+	if seeker, ok := reader.(io.ReadSeeker); ok && offset > 0 {
+		// File-based readers will maintain the read pointer.
+		// So for those readers calling Read func after a failure will resume the read rather than read from 0.
+		// We still explicitly seek if the reader supports it and needs resuming as we are not sure which type of readers are passed to this func
+		_, err := seeker.Seek(offset, 0)
+
+		if err != nil {
+			return fmt.Errorf("failed to seek to offset %d: %v, %w", offset, err, ErrNotRetryable)
+		}
+
+		// move progress bar to offset as well
+		bar.Add64(offset)
```

golangci-lint (1.64.8)

Error return value of `bar.Add64` is not checked

(errcheck)